### DISCUSSION TITLE
bot won't start , Error: Cannot find module 'mineflayer'

### DISCUSSION THREAD

**Discussion by Eugenir-774** [PrismarineJS/mineflayer#2818] [Q&A]
hello everyone, I recently started studying a java script for writing minecraft bots through mineflayer, in order to find out how it works, I downloaded several bots from the github open spaces, and when I try to run any, an error occurs:

Error: Cannot find module 'mineflayer'
Require stack:
- E:\minecraftbot\####-###\######\####.js
←[90m    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)←[39m
←[90m    at Module._load (node:internal/modules/cjs/loader:841:27)←[39m
←[90m    at Module.require (node:internal/modules/cjs/loader:1061:19)←[39m
←[90m    at require (node:internal/modules/cjs/helpers:103:18)←[39m
    at Object.<anonymous> (E:\minecraftbot\#######\##########\####.js:1:20)
←[90m    at Module._compile (node:internal/modules/cjs/loader:1159:14)←[39m
←[90m    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)←[39m
←[90m    at Module.load (node:internal/modules/cjs/loader:1037:32)←[39m
←[90m    at Module._load (node:internal/modules/cjs/loader:878:12)←[39m
←[90m    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: [ ←[32m'E:\\minecraftbot\\######\\########\\####.js'←[39m ]

please help, as far as I understand the main flyer and the plugins needed for the bots to work are downloaded,

**Comment by XingYuan55** [comment]
I have a similar situation #2920

**Comment by phsadhankar** [comment]
Try doing `npm i` in the terminal
